sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -757,4 +764,84 @@
+        return new PollerFlux<PhoneNumberSearch, PhoneNumberSearch>(pollInterval,
+            createSearchActivationOperation(options),
+            createSearchPollOperation(),
+            cancelSearchOperation(),
+            createSearchFetchResultOperation());
+    }
+
+    private Function<PollingContext<PhoneNumberSearch>, Mono<PhoneNumberSearch>> 
+        createSearchActivationOperation(CreateSearchOptions options) {
+        return (pollingContext) -> {
+            Mono<PhoneNumberSearch> response = createSearch(options).flatMap(createSearchResponse -> {
+                String searchId = createSearchResponse.getSearchId();
+                Mono<PhoneNumberSearch> phoneNumberSearch = getSearchById(searchId);
+                return phoneNumberSearch;
+            });
+            return response;
+        };
+    }
+
+    private Function<PollingContext<PhoneNumberSearch>, 
+        Mono<PollResponse<PhoneNumberSearch>>> createSearchPollOperation() {
+        return pollingContext ->
+            getSearchById(pollingContext.getActivationResponse().getValue().getSearchId())
+                .flatMap(getSearchResponse -> {
+                    SearchStatus status = getSearchResponse.getStatus();
+                    if (status.equals(SearchStatus.SUCCESS)) {
+                        return Mono.just(new PollResponse<>(
+                        LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, getSearchResponse));
+                    }
+                    if (status.equals(SearchStatus.EXPIRED)) {
+                        return Mono.just(new PollResponse<>(
+                        LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, getSearchResponse));
+                    }
+                    if (status.equals(SearchStatus.CANCELLED)) {
+                        return Mono.just(new PollResponse<>(
+                        LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, getSearchResponse));
+                    }
+                    if (status.equals(SearchStatus.RESERVED)) {
+                        return Mono.just(new PollResponse<>(
+                        LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, getSearchResponse));
+                    }
+                    if (status.equals(SearchStatus.ERROR)) {
+                        return Mono.just(new PollResponse<>(
+                        LongRunningOperationStatus.FAILED, getSearchResponse));
+                    }
+                    return Mono.just(new PollResponse<>(LongRunningOperationStatus.IN_PROGRESS, getSearchResponse));
+                });
+    }
+
+    private BiFunction<PollingContext<PhoneNumberSearch>,
+        PollResponse<PhoneNumberSearch>, Mono<PhoneNumberSearch>> 
+        cancelSearchOperation() {
+        return (pollingContext, firstResponse) -> {
+            cancelSearch(pollingContext.getActivationResponse().getValue().getSearchId());
+            return Mono.just(new PhoneNumberSearch());
```

Comment:
So if change the polling type as per my other comment, the return type would just change to Mono.just(SearchStatus.Cancelled) ..